tensorboard/webapp/metrics/effects/index.ts

```diff
@@ -195,23 +236,31 @@ export class MetricsEffects implements OnInitEffects {
     fetchInfos: CardFetchInfo[],
     experimentIds: string[]
   ) {
-    /**
-     * TODO(psybuzz): if 2 cards require the same data, we should dedupe instead of
-     * making 2 identical requests.
-     */
-    const requests: TimeSeriesRequest[] = fetchInfos.map((fetchInfo) => {
-      const {plugin, tag, runId, sample} = fetchInfo;
-      const partialRequest: TimeSeriesRequest = isSingleRunPlugin(plugin)
-        ? {plugin, tag, runId: runId!}
-        : {plugin, tag, experimentIds};
-      if (sample !== undefined) {
-        partialRequest.sample = sample;
-      }
-      return partialRequest;
-    });
-
     // Fetch and handle responses.
-    return of(requests).pipe(
+    return this.tagToEid$.pipe(
+      map((tagToEid): TimeSeriesRequest[] => {
+        const requests = fetchInfos.map((fetchInfo) => {
+          const {plugin, tag, runId, sample} = fetchInfo;
+          const filteredEids = experimentIds.filter((eid) =>
+            tagToEid[tag]?.has(eid)
+          );
+
+          const partialRequest: TimeSeriesRequest = isSingleRunPlugin(plugin)
+            ? {plugin, tag, runId: runId!}
+            : {plugin, tag, experimentIds: filteredEids};
+          if (sample !== undefined) {
+            partialRequest.sample = sample;
+          }
+          return partialRequest;
+        });
+        const uniqueRequests = new Set(
```

Comment:
Is there an actual problem you are trying to solve here? I don't see a test for this and I didn't see anything about it in the PR description.

Reply:
I am attempting to address the TODO left by psybuzz

Reply:
There was an existing test which verified this was doing the wrong thing which I updated. See the comment that I removed from line 375 of metrics_effects_test